pkg/cloudprovider/providers/aws/aws.go

```diff
@@ -2048,3 +1968,110 @@ func (s *AWSCloud) addFilters(filters []*ec2.Filter) []*ec2.Filter {
 	}
 	return filters
 }
+
+// Create a security group for a load balancer
+func (s *AWSCloud) createSecurityGroupForLoadBalancer(name string, vpcId string, ports []*api.ServicePort) (*string, error) {
+	sgName := "k8s-elb-" + name
+	sgDescription := "Security group for Kubernetes ELB " + name
+	securityGroupId, err := s.ensureSecurityGroup(sgName, sgDescription, vpcId)
+	if err != nil {
+		glog.Error("Error creating load balancer security group: ", err)
+		return nil, err
+	}
+
+	permissions := []*ec2.IpPermission{}
+	for _, port := range ports {
+		portInt64 := int64(port.Port)
+		protocol := strings.ToLower(string(port.Protocol))
+		sourceIp := "0.0.0.0/0"
+
+		permission := &ec2.IpPermission{}
+		permission.FromPort = &portInt64
+		permission.ToPort = &portInt64
+		permission.IpRanges = []*ec2.IpRange{{CidrIp: &sourceIp}}
+		permission.IpProtocol = &protocol
+
+		permissions = append(permissions, permission)
+	}
+	_, err = s.ensureSecurityGroupIngress(securityGroupId, permissions)
+	if err != nil {
+		return nil, err
+	}
+
+	return &securityGroupId, nil
+}
+
+// Creates a security group which allows ingress from all Kubernetes owned ELBs. This group is meant to be
+// attached to all node instances.
+func (s *AWSCloud) ensureSharedInstanceSecurityGroupForLoadBalancing(vpcId string, allInstances []*ec2.Instance) (*string, error) {
```

Comment:
It bites me every time, but we're supposed to use upper-case for acronyms in go i.e. `vpcID`.  I'm sure if you look around not-even-particularly-hard you can find some examples where I got this wrong ;-)